universal-login-sdk/lib/services/FutureWalletFactory.ts

```diff
@@ -0,0 +1,59 @@
+import {providers} from 'ethers';
+import {SupportedToken, ContractWhiteList} from '@universal-login/commons';
+import {createFutureWallet} from '../utils/counterfactual';
+import {BalanceObserver} from '../observers/BalanceObserver';
+import {DeploymentObserver} from '../observers/DeploymentObserver';
+import {BlockchainService} from './BlockchainService';
+
+type BalanceDetails = {
+  tokenAddress: string,
+  contractAddress: string
+};
+
+type FutureWallet = {
+  privateKey: string,
+  contractAddress: string,
+  waitForBalance: () => Promise<BalanceDetails>,
+  deploy: (ensName: string) => Promise<any>
+};
+
+export class FutureWalletFactory {
+
+  constructor(
+    private factoryAddress: string,
+    private provider: providers.Provider,
+    private supportedTokens: SupportedToken[],
+    private blockchainService: BlockchainService,
+    private contractWhiteList: ContractWhiteList,
+    private deployCallback: (publicKey: string, ensName: string) => Promise<string>) {
+  }
```

Comment:
To do:
1) `factoryAddress`, `supportedTokens`, `contractWhiteList` -> `config`
2) `deployCallback` -> `RelayerApi`

Next PR:

3) `provider` -> `ConterfacutalService`, `BlockchainService`

  `await createFutureWallet(this.factoryAddress, this.provider)` ->   
  `ConterfacutalService.createFutureWallet(this.factoryAddress)`

  `new BalanceObserver(this.supportedTokens, this.provider);` -> 
  `new BalanceObserver(this.supportedTokens, this.blockchainService);`

```ts
class ConterfacutalService {
  constructor(factoryAddress: string, provider: providers.Provider) {
  }
  createFutureWallet() {
  }
}
```

Reply:
✔️